src/Twilio/TwilioNoAuth.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Twilio.Clients;
+using Twilio.Exceptions;
+using Twilio.Clients.NoAuth;
+
+namespace Twilio
+{
+    /// <summary>
+    /// Default Twilio Client
+    /// </summary>
+    public class TwilioNoAuthClient
+    {
+
+        private static string _region;
+        private static string _edge;
+        private static TwilioNoAuthRestClient _restClient;
+        private static string _logLevel;
+
+        private TwilioNoAuthClient() { }
+
+        /// <summary>
+        /// Get the rest client
+        /// </summary>
+        /// <returns>The rest client</returns>
+        public static TwilioNoAuthRestClient GetRestClient()
+        {
+            if (_restClient != null)
+            {
+                return _restClient;
+            }
+
+            _restClient = new TwilioNoAuthRestClient(region: _region, edge: _edge)
```

Comment:
If customer has set region and edge in **TwilioRestClient**, then NoAuthClient call should also go to provided region and edge.

Reply:
we might be able to use no auth client, but even when customer only wants to do a rest call without auth, he/she should pass username and password. is this fine? if yes we can remove the no auth class